src/main/java/mate/academy/spring/service/CustomUserDetailService.java

```diff
@@ -0,0 +1,38 @@
+package mate.academy.spring.service;
+
+import static org.springframework.security.core.userdetails.User.withUsername;
+
+import java.util.Optional;
+import mate.academy.spring.model.User;
+import org.springframework.security.core.userdetails.User.UserBuilder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CustomUserDetailService implements UserDetailsService {
+    private final UserService userService;
+
+    public CustomUserDetailService(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+        Optional<User> userOptional = userService.findByEmail(email);
+        User user = userOptional.orElseThrow(() ->
+                new RuntimeException("Can't find user by email " + email));

```

Comment:
Be careful when you check user for null in CustomUserDetailsService, can user be null, when we get him using Optional.get() or Optional.orElseThrow()?